src/custom/components/SearchModal/CurrencyList/index.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { CurrencyAmount } from '@uniswap/sdk'
+
+import { LONG_PRECISION } from 'custom/constants'
+
+import CurrencyListMod, { StyledBalanceText } from './CurrencyListMod'
+
+export function Balance({ balance }: { balance: CurrencyAmount }) {
+  return <StyledBalanceText title={balance.toExact()}>{balance.toSignificant(LONG_PRECISION)}</StyledBalanceText>
+}
+
+export default function CurrencyList(
+  ...params: Parameters<typeof CurrencyListMod>
+): ReturnType<typeof CurrencyListMod> {
+  return <CurrencyListMod {...params[0]} BalanceComponent={Balance} />
```

Comment:
hard to read :see_no_evil: but fine if it works idgaf :D

Reply:
Any suggestion?

Reply:
https://github.com/gnosis/gp-swap-ui/pull/386